mantidproject/mantid
# Add legend properties to the plot script generator

**Description of work.**

The python script generated from a plot will now reproduce all of the legend settings. The script will only contain legend properties that are different to the matplotlib default values.

**To test:**

- Load a workspace and plot a spectrum (or multiple spectra on the same plot).
- Open the "Figure Options" dialog by clicking the gear icon and navigate to the Legend tab.
- Play around with all of the settings (including those inside "Advanced Options").
- Apply and close the dialog.
- Click the generate script button and select "Script to clipboard".
- Close the plot.
- Paste the generated script into the python editor widget, and run the script.
- The plot, including all legend formatting, should be reproduced.

Additionally, repeat the above, but without changing any of the legend settings. The generated script should not contain any legend properties.


Fixes #27397.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
